Marlin/src/inc/Conditionals_post.h

```diff
@@ -2860,6 +2860,11 @@
   #define HAS_MOTOR_CURRENT_PWM 1
 #endif
 
+// Add features that need hardware PWM here
+#if ANY(FAST_PWM_FAN, SPINDLE_LASER_USE_PWM, MOTOR_CURRENT_PWM_FEATURE, HAS_MOTOR_CURRENT_PWM)
+  #define NEEDS_HARDWARE_PWM 1
+#endif
+
```

Comment:
HAS_LCD_BRIGHTNESS can be pwm... may need some flag more explicit (for both lcd & tft)

Reply:
Strange. How did it ever work prior to this PR ?

Reply:
I don't think it would have worked correctly prior to these changes.
STM and LCP176x HAL do not condition on NEEDS_HARDWARE_PWM, it's always on.
I will add it.